pkg/core/resources/store/pagination_store.go

```diff
@@ -0,0 +1,91 @@
+package store
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/registry"
+)
+
+func NewPaginationStore(delegate ResourceStore) ResourceStore {
+	return &PaginationStore{
+		delegate: delegate,
+	}
+}
+
+type PaginationStore struct {
+	delegate ResourceStore
+}
+
+func (p *PaginationStore) Create(ctx context.Context, resource model.Resource, optionsFunc ...CreateOptionsFunc) error {
+	return p.delegate.Create(ctx, resource, optionsFunc...)
+}
+
+func (p *PaginationStore) Update(ctx context.Context, resource model.Resource, optionsFunc ...UpdateOptionsFunc) error {
+	return p.delegate.Update(ctx, resource, optionsFunc...)
+}
+
+func (p *PaginationStore) Delete(ctx context.Context, resource model.Resource, optionsFunc ...DeleteOptionsFunc) error {
+	return p.delegate.Delete(ctx, resource, optionsFunc...)
+}
+
+func (p *PaginationStore) Get(ctx context.Context, resource model.Resource, optionsFunc ...GetOptionsFunc) error {
+	return p.delegate.Get(ctx, resource, optionsFunc...)
+}
+
+func (p *PaginationStore) List(ctx context.Context, list model.ResourceList, optionsFunc ...ListOptionsFunc) error {
```

Comment:
Optimisation:
if the `FilterFunc` is nil and `PageSize` is 0 and `PageOffset` is `""` just use delegate to original store.

For the most cases in our code we don't need pagination and filtering